### ISSUE TITLE
Python: Enum argument missing if it shares value with another

### ISSUE THREAD

**Issue by JohanLorenzo** [sphinx-doc/sphinx#5139]
Thank you for this great library!

Subject: All members of a python Enum aren't rendered if some of them are sharing the same value

### Problem

Let this Python enum:

```py
class MyEnum(Enum):
    """Some docstring"""

    A = 1
    B = 2
    C = 3
    C_ = 3
```

Sphinx 1.7.5 with the autodoc extension generates:

```
 class my_module.MyEnum

    Bases: enum.Enum

    Some docstring

    B = 2
    C = 3
    A = 1
```

Another scenario:

```py
class MyEnum(Enum):
    """Some docstring"""

    A = 1
    B = 2
    C = 3
    C_ = 3       #:doc
```

gets generated into:

```
 class my_module.MyEnum

    Bases: enum.Enum

    Some docstring

    B = 2
    C_ = None
       doc
    C = 3
    A = 1
```

#### Procedure to reproduce the problem
```
make html
```
Makefile being generated by sphinx-quickstart 1.7.5.

#### Error logs / results
See above

#### Expected results

```
 class my_module.MyEnum

    Bases: enum.Enum

    Some docstring

    A = 1
    B = 2
    C = 3
    C_ = 3
```

### Reproducible project / your project
- https://github.com/mozilla-releng/mozilla-version/blob/46c1208623dd5fc78f69596ec05a06d87fad2e40/mozilla_version/version.py#L26
- https://mozilla-version.readthedocs.io/en/latest/mozilla_version.html#module-mozilla_version.version

### Environment info
- OS: Latest Arch Linux
- Python version: 3.6.5
- Sphinx version: 1.7.5

**Comment by tk0miya** [comment]
It seems `C_` is defined as a kind of alias of `C`.
```
$ python
Python 3.6.5 (default, Mar 31 2018, 01:44:52)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import Enum
>>> class MyEnum(Enum):
...     A = 1
...     B = 2
...     C = 3
...     C_ = 3
...
>>> MyEnum.C
<MyEnum.C: 3>
>>> MyEnum.C_
<MyEnum.C: 3>
```

But, unfortunately, it is not listed on result of `dir()`. So we have to find the way to fetch such alias definition.
```
>>> dir(MyEnum)
['A', 'B', 'C', '__class__', '__doc__', '__members__', '__module__']
```

**Comment by tk0miya** [comment]
Python docs says `__members__` attribute can access these aliases!

>The special attribute __members__ is an ordered dictionary mapping names to members. It includes all names defined in the enumeration, including the aliases:
https://docs.python.org/3/library/enum.html#iteration

**Comment by tk0miya** [comment]
Fixed by #5171.
Thank you for reporting.

**Comment by JohanLorenzo** [comment]
Thank you for fixing it this quick! :D
